test/tools/OpenCover/OpenCover.psm1

```diff
@@ -0,0 +1,418 @@
+        [Parameter(Mandatory=$true,Position=1,ParameterSetName="file")][string]$RunFile2,
+        [Parameter(Mandatory=$true,Position=0,ParameterSetName="coverage")][Object]$Run1, 
+        [Parameter(Mandatory=$true,Position=1,ParameterSetName="coverage")][Object]$Run2
+        )
+
+    if ( $PSCmdlet.ParameterSetName -eq "file" ) 
+    {
+        [string]$xmlPath1 = (get-item $Run1File).Fullname        
+        $Run1 = (Get-CoverageData -xmlPath $xmlPath1)
+
+        [string]$xmlPath2 = (get-item $Run1File).Fullname
+        $Run2 = (Get-CoverageData -xmlPath $xmlPath2)
+    }
+    
+    (Get-CodeCoverageChange -r1 $Run1 -r2 $Run2)
+}
+
+## We are taking dependency on private build of OpenCover till a new release is available.
+## This is required for collecting code coverage numbers with portable PDBs.  
+<#
+.Synopsis
+   Install OpenCover by downloading the 4.6.589 version.  
+.Description
+   Install OpenCover version 4.6.589. This version supports debugType as portable.
+#>
+function Install-OpenCover
+{
+    param ( 
+        [parameter()][string]$version = "4.6.589",
+        [parameter()][string]$targetDirectory = $PWD,
+        [parameter()][switch]$force
+        )
+
+    $webclient = [System.Net.WebClient]::New()
+    $filename =  "opencover.${version}.zip"
+    $packageUrl = "https://ci.appveyor.com/api/buildjobs/xj78v6dac42uob8q/artifacts/main%2Fbin%2Fzip%2Fopencover.4.6.589.zip"
+    if ( test-path $PWD/$Filename ) 
+    {
+        if ( $force ) 
+        {
+            remove-item -force "$PWD/$Filename"
+        }
+        else 
+        {
+            throw "package already exists, not downloading"
+        }
+    }
+    if ( test-path "$targetDirectory/OpenCover" ) 
+    {
+        if ( $force ) 
+        {
+            remove-item -recurse -force "$targetDirectory/OpenCover"
+        }
+        else 
+        {
+            throw "$targetDirectory/OpenCover exists"
+        }
+    }
+
+    $webclient.DownloadFile($packageUrl, "$PWD/$filename")
+    if ( ! (test-path $filename) ) 
+    {
+        throw "Download failed: $packageUrl"
+    }
+    
+    import-module Microsoft.PowerShell.Archive
+    Expand-ZipArchive -Path "$PWD/$filename" -DestinationPath "$targetDirectory/OpenCover"
+}
+
+<#
+.Synopsis
+   Invoke-OpenCover runs tests under OpenCover to collect code coverage.
+.Description
+   Invoke-OpenCover runs tests under OpenCover by executing tests on PowerShell.exe located at $PowerShellExeDirectory.
+.EXAMPLE
+   Invoke-OpenCover -TestDirectory $pwd/test/powershell -PowerShellExeDirectory $pwd/src/powershell-win-core/bin/debug/netcoreapp1.0/win10-x64 
+#>
+function Invoke-OpenCover
+{
+    [CmdletBinding(SupportsShouldProcess=$true)]
+    param ( 
+        [parameter()]$OutputLog = "$pwd/OpenCover.xml",
+        [parameter(Mandatory=$true)]$TestDirectory,
+        [parameter()]$OpenCoverPath = "$HOME\AppData\Local\Apps\OpenCover", # this is the default install location for open cover
```

Comment:
How is `$env:LOCALAPPDATA` (and you should use that instead of `$HOME\AppData\Local`) the default? I tried `Install-OpenCover` and it used `$pwd`.